### PR TITLE
chore: bump typescript-eslint to v8 with project service

### LINKED ISSUE #5867: feat: use typescript-eslint@v8 with project service
### Describe the feature you'd like to request

Following #4540 & coming over from https://github.com/typescript-eslint/typescript-eslint/issues/9141: I'm back again to try out the next typescript-eslint major version on trpc! 🙌 

The big new change we have this time is the new [project service (`parserOptions.projectService`)](https://typescript-eslint.io/blog/announcing-typescript-eslint-v8-beta#project-service). It should result in faster typed linting with an easier configuration.

```diff
parserOptions: {
-  project: [
-    // ...
-  ],
+  projectService: true,
  tsconfigRootDir: import.meta.dirname,
},
```

### Describe the solution you'd like to see

I'll send a draft PR with the changes.

### Describe alternate solutions

I suppose trpc could stay on typescript-eslint@v6, or use its own bespoke lint configs... neither option seems very appealing to me. 😄

### Additional information

_No response_

### 👨‍👧‍👦 Contributing

- [X] 🙋‍♂️ Yes, I'd be down to file a PR implementing this feature!

### PR DESCRIPTION
**Pull request by JoshuaKGoldberg**
Closes #5867

## 🎯 Changes

1. Updates `typescript-eslint` to the latest v8 beta, `rc-v8`
    * Also `eslint` to typescript-eslint's minimum supported version, and `prettier` to deduplicate shared dependency versions
2. Enables the new [`parserOptions.projectService`](https://typescript-eslint.io/blog/announcing-typescript-eslint-v8-beta#project-service), simplifying the ESLint config a bit
3. For any rule newly enabled via the preset configs that produced a lot of errors, disabled under a TODO in the ESLint config

Note that this is still a beta version of typescript-eslint@v8. So unless you want to live on the _edge_, I'd suggest waiting until we hit stable this summer to merge.

## ✅ Checklist

- [x] I have followed the steps listed in the [Contributing guide](https://github.com/trpc/trpc/blob/main/CONTRIBUTING.md).
- [x] If necessary, I have added documentation related to the changes made.
- [x] I have added or updated the tests related to the changes made.

### CODE REVIEW

File: examples/next-prisma-starter/src/server/env.js [hunk 1]
```diff
@@ -3,7 +3,7 @@
  * This file is included in `/next.config.js` which ensures the app isn't built with invalid env vars.
  * It has to be a `.js`-file to be imported there.
  */
-/* eslint-disable @typescript-eslint/no-var-requires */
+/* eslint-disable @typescript-eslint/no-require-imports */
```
Comment (JoshuaKGoldberg) [inline examples/next-prisma-starter/src/server/env.js:6]: https://github.com/typescript-eslint/typescript-eslint/pull/8334: we merged the two rules into the one `no-require-imports`.

File: examples/next-prisma-todomvc/src/pages/[filter].tsx [hunk 2]
```diff
@@ -221,6 +221,7 @@ export default function TodosPage(props: PageProps) {
 
           <input
             className="new-todo"
+            // eslint-disable-next-line @typescript-eslint/no-unnecessary-type-assertion
             placeholder={t('what_needs_to_be_done') as string}
```
Comment (JoshuaKGoldberg) [inline examples/next-prisma-todomvc/src/pages/[filter].tsx:225]: This one stumped me. I couldn't reproduce it in a standalone repro: https://github.com/JoshuaKGoldberg/repros/tree/next-18n-and-type-assertion.

...are there some custom fancy types getting added? I time boxed investigating and couldn't find any.

File: examples/next-sse-chat/src/app/channels/[channelId]/chat.tsx [hunk 3]
```diff
@@ -55,6 +55,7 @@ export function Chat(props: Readonly<{ channelId: string }>) {
                     livePosts.query.isFetchingNextPage
                   }
                   onClick={() => {
+                    // eslint-disable-next-line @typescript-eslint/no-floating-promises
                     livePosts.query.fetchNextPage();
```
Comment (JoshuaKGoldberg) [inline examples/next-sse-chat/src/app/channels/[channelId]/chat.tsx:59]: I don't know why this wasn't flagging before. But it's a legit rule violation.

File: packages/next/src/ssrPrepass.ts [hunk 4]
```diff
@@ -59,7 +59,7 @@ export const ssrPrepass: TRPCPrepassHelper = (opts) => {
       if (typeof parent.ssr === 'function') {
         try {
           return await parent.ssr({ ctx: appOrPageCtx.ctx });
-        } catch (e) {
+        } catch {
```
Comment (JoshuaKGoldberg) [inline packages/next/src/ssrPrepass.ts:62]: As of https://github.com/eslint/eslint/issues/17974 -> https://github.com/eslint/eslint/pull/18043, these are now flagged by default by `no-unused-vars`. They were unused before.

File: packages/tests/server/react/__testHelpers.tsx [hunk 5]
```diff
@@ -133,7 +133,7 @@ export function createAppRouter() {
         const last = items[items.length - 1];
         const nextIndex = db.posts.findIndex((item) => item === last) + 1;
         if (db.posts[nextIndex]) {
-          nextCursor = db.posts[nextIndex]!.createdAt;
+          nextCursor = db.posts[nextIndex].createdAt;
```
Comment (JoshuaKGoldberg) [inline packages/tests/server/react/__testHelpers.tsx:136]: I don't know which bug was fixed here, but this is a legit unnecessary `!`.

**Comment by KATT** [comment]
I got a big branch that will conflict with this (#5865) so hope the changes are mainly automatic

**Comment by JoshuaKGoldberg** [comment]
Oh fun!

Yes, these are mostly automatic. I don't mind recreating as needed. 🙂

**Comment by Nick-Lucas** [comment]
Hey @JoshuaKGoldberg this has been idle for quite a long time so I'm going to close it out. The contribution would be very welcome though if you want to re-open

**Pull request closed without merging**